packages/kbn-config-schema/src/types/object_type.ts

```diff
@@ -77,7 +100,54 @@ export class ObjectType<P extends Props = any> extends Type<ObjectResultType<P>>
       .options({ stripUnknown: { objects: unknowns === 'ignore' } });
 
     super(schema, typeOptions);
-    this.props = schemaKeys;
+    this.props = props;
+    this.propSchemas = schemaKeys;
+    this.options = options;
+  }
+
+  /**
+   * Return a new `ObjectType` instance extended with given `newProps` properties.
```

Comment:
Could you add a comment that it doesn't support a deep extend?

Reply:
Yea, I will. Maybe important to note that this is a limitation than the `getRaw` approach did not have, as we would have been able to do something like

```ts
const origin = schema.object({
   initial: schema.string(),
  nested: schema.object({
    a: schema.string(),
    b: schema.string()
  })
});

const extended = schema.object({
   ...origin.getRaw(),
   nested: schema.object({
      ...origin.getRaw().nested.getRaw(),
      c: schema.string()
   })
});
```

Reply:
>  Maybe important to note that this is a limitation than the getRaw approach did not have,

The current solution supports this functionality, but it requires a nested object schema to be exported as well.

```js
const nested = schema.object({
  a: schema.string(),
  b: schema.string()
});

const origin = schema.object({
  initial: schema.string(),
  nested,
});

const extended = origin.extend({
  nested: nested.extend({
    c: schema.string()
  })
});
```

Reply:
Yea, but the limitation is that the original schema's owner should export each individual nested schema, which is not minor imho.

But you're right, there is a workaround for that so it's better than nothing.